helm-charts/support/templates/prometheus-htpasswd.yaml

```
@@ -0,0 +1,10 @@
+
+{{- if .Values.prometheusIngressAuthSecret.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: prometheus-ingress-auth-basic
+type: Opaque
+data:
+  auth: {{ htpasswd .Values.prometheusIngressAuthSecret.username .Values.prometheusIngressAuthSecret.password | b64enc | quote}}
+{{- end }}
```

- Remove blank new line in the beginning of file
- Add new line in end of file
- Add right whitespace chomp on the helm template expression before the first content has been rendered as a rule of thumb to ensure we don't render a blank line as the first thing as our output of the rendered helm template.

```suggestion
{{- if .Values.prometheusIngressAuthSecret.enabled -}}
apiVersion: v1
kind: Secret
metadata:
  name: prometheus-ingress-auth-basic
type: Opaque
data:
  auth: {{ htpasswd .Values.prometheusIngressAuthSecret.username .Values.prometheusIngressAuthSecret.password | b64enc | quote}}
{{- end }}

```

---

Done!